=== modified file 'l10n_es_extras/l10n_ES_remesas/wizard/export_remesas.py'
--- l10n_es_extras/l10n_ES_remesas/wizard/export_remesas.py	2009-09-14 16:04:40 +0000
+++ l10n_es_extras/l10n_ES_remesas/wizard/export_remesas.py	2009-11-02 15:34:27 +0000
@@ -385,7 +385,7 @@
         return {'note':log(), 'reference':orden.id, 'pay':False, 'state':'failed'}
     else:
         file = base64.encodestring(txt_remesa)
-        fname = (_('remesa') + '_' + orden.mode.tipo + '_' + orden.reference + '.txt').replace('/','-')
+        fname = (_('remesa') + '_' + orden.mode.tipo + '_' + orden.reference + '.q58').replace('/','-')
         pool.get('ir.attachment').create(cr, uid, {
             'name': _('Remesa ') + orden.mode.tipo + ' ' + orden.reference,
             'datas': file,

