=== modified file 'l10n_es_extras/l10n_ES_remesas/wizard/export_remesas.py'
--- l10n_es_extras/l10n_ES_remesas/wizard/export_remesas.py	2009-11-05 09:18:27 +0000
+++ l10n_es_extras/l10n_ES_remesas/wizard/export_remesas.py	2010-02-04 08:54:06 +0000
@@ -365,7 +365,7 @@
         texto = '5870'
         texto += (orden.mode.bank_id.partner_id.vat[2:] + orden.mode.sufijo).zfill(12)
         texto += 72*' '
-        totalordenante = int(-orden.total * 100)
+        totalordenante = int(round(-orden.total * 100,0))
         texto += str(totalordenante).zfill(10)
         texto += 6*' '
         texto += str(num_recibos).zfill(10)

