=== modified file 'sale/sale.py'
--- sale/sale.py	2009-09-22 06:42:56 +0000
+++ sale/sale.py	2009-09-24 15:58:19 +0000
@@ -374,7 +374,7 @@
             'account_id': a,
             'partner_id': order.partner_id.id,
             'address_invoice_id': order.partner_invoice_id.id,
-            'address_contact_id': order.partner_invoice_id.id,
+            'address_contact_id': order.partner_order_id.id,
             'invoice_line': [(6, 0, lines)],
             'currency_id': order.pricelist_id.currency_id.id,
             'comment': order.note,

