=== modified file 'base_vat/partner.py'
--- base_vat/partner.py	2009-06-23 08:54:41 +0000
+++ base_vat/partner.py	2009-10-06 07:51:15 +0000
@@ -379,7 +379,7 @@
             23: 'E',
         }
 
-        if vat[0] in ('A', 'B', 'C', 'D', 'E', 'F', 'G', 'H'):
+        if vat[0] in ('A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'J', 'U', 'V'):
             try:
                 int(vat[1:])
             except:
@@ -394,7 +394,7 @@
             if check != int(vat[8]):
                 return False
             return True
-        elif vat[0] in ('N', 'P', 'Q', 'S'):
+        elif vat[0] in ('N', 'P', 'Q', 'R', 'S', 'W'):
             try:
                 int(vat[1:8])
             except:
@@ -408,12 +408,19 @@
             if check != vat[8]:
                 return False
             return True
-        elif vat[0] in ('K', 'L', 'M', 'X'):
+        elif vat[0] in ('K', 'L', 'M', 'X', 'Y', 'Z'):
+            if vat[0] == 'Y':
+                check_value = '1' + vat[1:8]
+            elif vat[0] == 'Z':
+                check_value = '2' + vat[1:8]
+            else:
+                check_value = vat[1:8]
+
             try:
-                int(vat[1:8])
-            except:
+               int(check_value)
+            except ValueError:
                 return False
-            check = 1 + (int(vat[1:8]) % 23)
+            check = 1 + (int(check_value) % 23)
 
             check = conv[check]
             if check != vat[8]:

